query-graphs-js/src/graphPath.ts

```diff
@@ -941,11 +1004,6 @@ function advancePathWithNonCollectingAndTypePreservingTransitions<TTrigger, V ex
       }
       excludedEdges = addEdgeExclusion(excludedEdges, edge);
 
-      if (isConditionExcluded(edge.conditions, excludedConditions)) {
```

Comment:
Note to reviewer: this condition hasn't been removed, just moved down. The reason I did that is that this condition is a tad more expansive to check than the following ones and they can be swapped easily, so having that one later is probably a tad more efficient. In reality though, this probably don't make much of a difference, especially after the other optimisations of this commit, but I had made that change early and figured "why change back now?".